flask_smorest/etag.py

```diff
@@ -250,9 +250,30 @@ def _prepare_etag_doc(self, doc, doc_info, *, app, method, **kwargs):
             method_u = method.upper()
             if method_u in self.METHODS_CHECKING_NOT_MODIFIED:
                 responses[304] = http.HTTPStatus(304).name
+                doc.setdefault("parameters", []).append({
+                    'name': 'If-None-Match',
+                    'in': 'header',
+                    'description': 'Tag to check against',
+                    'schema': {'type': 'string'}})
             if method_u in self.METHODS_NEEDING_CHECK_ETAG:
                 responses[412] = http.HTTPStatus(412).name
                 responses[428] = http.HTTPStatus(428).name
+                doc.setdefault("parameters", []).append({
+                    'name': 'If-Match',
+                    'in': 'header',
+                    'required': True,
+                    'description': 'Tag to check against',
+                    'schema': {'type': 'string'}})
+            if method_u in self.METHODS_ALLOWING_SET_ETAG:
+                success_status_code = doc_info.get('success_status_code')
+                if success_status_code is not None:
```

Comment:
Branch coverage fails here. Do you want me to add a new test case without `@blp.response(204)`?